mozci/push.py

```diff
@@ -351,7 +359,21 @@ def add(result):
 
             normalized_tasks.append(task)
 
-        return [Task.create(**task) for task in normalized_tasks]
+        tasks = [Task.create(**task) for task in normalized_tasks]
+        self.cache_data(tasks)
+        return tasks
+
+    def cache_data(self, tasks):
+        push_tasks = {}
+        for task in tasks:
```

Comment:
I had considered iterating here over tasks that are TestTask instances and call `_load_errorsummary()`.
It was very slow. We could parallelize it if that is what you prefer.

Reply:
An alternative to parallelization that could work is fetching all tasks as a group as I do in [Treeherder](https://github.com/mozilla/treeherder/blob/master/treeherder/etl/management/commands/ingest.py#L95-L109).

Reply:
I mentioned it in another comment, we already have some code in push.py to do that, so you could reuse it

Reply:
The treeherder code is loading all tasks from a group of tasks, but here it wouldn't help: we already have the tasks of the group (which are the same as the tasks of the push).

Reply:
N.B.: a `group` in mozci is a manifest, a `group` in Taskcluster is a group of tasks